shared/utils/relabel-ids.py

```diff
@@ -187,19 +187,21 @@ def _add_cce_id_refs_to_oval_checks(self, idmappingdict):
                 continue
 
             xccdfcceid = idmappingdict[ovalid]
-            if not ssgcommon.cce_is_valid(xccdfcceid):
-                msg = "The CCE ID {0} is not in valid form.".format(xccdfcceid)
-                raise ssgcommon.SSGError(msg)
-
-            # Then append the <reference source="CCE" ref_id="CCE-ID" /> element right
-            # after <description> element of specific OVAL check
-            ccerefelem = ssgcommon.ElementTree.Element(
-                'reference', ref_id=xccdfcceid, source="CCE")
-            ovaldesc.addnext(ccerefelem)
-            # Sanity check if appending succeeded
-            if ccerefelem.getprevious() is not ovaldesc:
-                msg = "Failed to add CCE ID to {0}.".format(ovalid)
-                raise ssgcommon.SSGError(msg)
+            if ssgcommon.cce_is_valid(xccdfcceid):
+                # Then append the <reference source="CCE" ref_id="CCE-ID" /> element right
+                # after <description> element of specific OVAL check
+                ccerefelem = ssgcommon.ElementTree.Element(
+                    'reference', ref_id=xccdfcceid, source="CCE")
+                metadata = rule.find(".//{%s}metadata" % self.CHECK_NAMESPACE)
+                metadata.append(ccerefelem)
+                # Sanity check if appending succeeded
+                cce_ref = None
+                for ref in metadata.findall(".//reference"):
+                    cce_ref = ref if ref.attrib.get("ref_id") == xccdfcceid else None
```

Comment:
I am sorry for endlessly nitpicking around this loop but IMO the logic is still flawed.

take this case:
first reference is non-cce
second reference is cce
third reference is non cce

the final cce_ref will be None, this certainly is not what you are looking for.

you need to set cce_ref **if and only if** the ref_id is cceid

Reply:
actually I think this whole block is quite ridiculous, we are checking that elementtree did what it was supposed to do...

Reply:
@mpreisler I agree with that statement as well. I thought it was really odd. Removed the sanity checking block as it seems.... unneeded.